core/files/uploads-manager.php

```diff
@@ -489,9 +489,18 @@ private function add_file_extension_to_allowed_extensions_list( $file_type ) {
 	 * @access private
 	 *
 	 * @param $file
+	 * @param array|null $allowed_file_extensions
+	 *
 	 * @return array|\WP_Error
 	 */
-	private function save_base64_to_tmp_file( $file ) {
+	private function save_base64_to_tmp_file( $file, $allowed_file_extensions = null ) {
+		$file_extension = pathinfo( $file['fileName'], PATHINFO_EXTENSION );
```

Comment:
Maybe use `wp_check_filetype()`

Reply:
I do not think in this case it will change much, as we don't analyze the content, but only the file name.